lib/resty/websocket/client.lua

```diff
@@ -98,9 +98,17 @@ function _M.connect(self, uri, opts)
         path = "/"
     end
 
-    local ssl_verify, proto_header, origin_header, sock_opts = false
+    local ssl_verify, proto_header, origin_header, custom_header, sock_opts = false
 
     if opts then
+        local headers = opts.headers
+        if headers then
+            custom_header = ""
+            for field,value in pairs(headers) do
```

Comment:
Although this is not the most performant way it maintains the usual design of other lua libraries APIs, like [lua-resty-http](https://github.com/pintsized/lua-resty-http/blob/master/lib/resty/http.lua#L280) does for headers, which I also use in resty projects, and different APIs design across libraries usually generates more undesired work.